qt/python/mantidqt/dialogs/spectraselectordialog.py

```diff
@@ -227,6 +336,171 @@ def _is_input_valid(self):
         return self.selection is not None
 
 
+AdvancedPlottingOptionsWidgetUI, AdvancedPlottingOptionsWidgetUIBase = load_ui(__file__, 'advancedplotswidget.ui')
+
+
+class AdvancedPlottingOptionsWidget(AdvancedPlottingOptionsWidgetUIBase):
+    def __init__(self, parent=None):
+        super(AdvancedPlottingOptionsWidget, self).__init__(parent=parent)
+        ui = AdvancedPlottingOptionsWidgetUI()
+        ui.setupUi(self)
+
+        ui.logs_valid.setIcon(red_asterisk())
+        ui.logs_valid.hide()
+
+        ui.log_value_combo_box.currentTextChanged.connect(self._log_value_changed)
+        ui.error_bars_check_box.clicked.connect(self._toggle_errors)
+        ui.custom_log_line_edit.textEdited.connect(self._validate_custom_logs)
+        ui.plot_axis_label_line_edit.textEdited.connect(self._axis_name_changed)
+
+        self.ui = ui
+        self._parent = parent
+        self._populate_log_combo_box()
+        ui.plot_axis_label_line_edit.setText(ui.log_value_combo_box.currentText())
+
+    def _log_value_changed(self, text: str) -> None:
+        # Don't need to do anything while the combo box is being populated.
+        if not self._parent._ui:
```

Comment:
I think this will currently just return if `self` is anything that has a parent that's not `_ui`... should it be?
```suggestion
        if self._parent._ui:
```

Reply:
The reason that if-statement is there is because when `_populate_log_combo_box` is executing, the `log_value_combo_box.currentTextChanged` signal will be emitted multiple times, which calls the `_log_value_changed` function. At this point the GUI is just being set up so there's no need to perform any of the actions in this function, and so there's a check for the parent's `_ui` attribute, which is initially set to None, no longer being None, at which point we know the user is actually interacting with the GUI and it's not just setup.

Reply:
Got you :+1: